source/extensions/transport_sockets/tls/context_impl.h

```diff
@@ -101,6 +101,14 @@ class ContextImpl : public virtual Envoy::Ssl::Context {
 
   std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> getPrivateKeyMethodProviders();
 
+  // Called by verifyCallback to do the actual cert chain verification.
+  int doVerifyCertChain(X509_STORE_CTX* store_ctx, Ssl::SslExtendedSocketInfo* ssl_extended_info,
```

Comment:
This function is named `doVerifyCertChain` but it only takes a leaf cert. I would change this to also take the cert chain (whether the leaf cert and cert chain are passed in separately or together, and whether they're X509 objects or just byte buffers are up to you).

The implementation of this function calls verifyCertificate, which calls matchSubjectAltName, so the SAN processing code in EnvoyQuicProofVerifier::VerifyCertChain is unnecessary.